src/pandas_openscm/index_manipulation.py

```diff
@@ -841,7 +842,7 @@ def set_levels(
             new_level, new_codes = create_level_from_collection(level, value)
         else:
             new_level = pd.Index([value], name=level)
-            new_codes = list(np.zeros(ini.shape[0]))
+            new_codes = np.zeros(ini.shape[0], dtype=int)
```

Comment:
Just dropping the cast to a list here as it isn't needed (the specification of the dtype wasn't something I expected, but it makes sense as the default is float, which isn't 100% correct)